### PR TITLE
Cleanup `subspan_view`

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Turns things inside out to simplify code significantly.

Also includding a cherry-picked commit from #1861 fixing a stride issue.

### CODE REVIEW

**Comment by SimonHeybrock** [comment]
@jl-wynen The diff is probably hard to read, but if you go commit-by-commit it should be much easier to follow. Or you trust the tests.

File: variable/subspan_view.cpp [hunk 1]
```diff
@@ -4,113 +4,53 @@
...
-
 template <class T>
 auto make_subspans(T *base, const Variable &indices,
                    const scipp::index stride) {
-  const auto spans = variable::transform<scipp::index_pair>(
+  if (stride != 1)
+    throw std::logic_error(
+        "span only supports stride=1, this should be "
+        "unreachable due to an earlier check, may want to generalize this "
+        "later to support in particular stride=0 for broadcasted buffers");
```
Comment (jl-wynen) [inline variable/subspan_view.cpp:23]: A bit of a strange error message.

**Review by jl-wynen — COMMENTED** [review]
I am not sure what exactly happened here. But the new implementation at least looks neater than the old.

File: variable/subspan_view.cpp (hunk 1 above)
Comment (SimonHeybrock) [reply variable/subspan_view.cpp:23]: It is not user facing, just as an extra safety check.

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**